runtime/src/packet.rs

```diff
@@ -25,7 +25,7 @@ pub struct Packet {
     pub len: usize, // Length in bytes
 }
 
-const MAX_PAYLOAD_SIZE: usize = 2050; // in dwords
+const MAX_PAYLOAD_SIZE: usize = 3586; // in dwords
```

Comment:
We'd probably want to test this against any new stack overflow detection features being developed.

Reply:
Agreed.